homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,143 @@
+"""The myIO integration."""
+from datetime import timedelta
+import logging
+
+from myio.comms_thread import CommsThread  # pylint: disable=import-error
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import CONF_PORT_APP, CONF_REFRESH_TIME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_NAME, default="myIO-Server"): str,
+                vol.Required(CONF_HOST, default="192.168.1.170"): str,
+                vol.Required(CONF_PORT, default="80"): int,
+                vol.Required(CONF_PORT_APP, default="843"): int,
+                vol.Required(CONF_USERNAME, default="admin"): str,
+                vol.Required(CONF_PASSWORD, default="admin"): str,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+COMMS_THREAD = CommsThread()
+
+
+async def async_setup(hass, config):
+    """Set up the myIO-server component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up config entry."""
+    _LOGGER.debug(DOMAIN)
+    _server_name = slugify(config_entry.data[CONF_NAME])
+    _server_status = "Offline"
+    hass.data[_server_name] = {}
+    hass.states.async_set(f"{_server_name}.state", _server_status)
```

Comment:
Don't set states like this. Create entities instead.

Reply:
I deleted this line, and solved my status checker in the hass.data database...